jmix-cuba/cuba/src/main/java/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java

```diff
@@ -2307,7 +2307,16 @@ protected void fillTableActions() {
             addToCurSetBtn.setAction(addToCurrSet);
             table.addAction(addToCurrSet);
 
-            removeFromCurrSet = new RemoveFromSetAction(table);
+            if(table instanceof ListComponent) {
+                // CUBA-table case
+                removeFromCurrSet = new RemoveFromSetAction(table);
+            } else {
+                // Jmix-table case
+                JmixRemoveFromSetAction action = new JmixRemoveFromSetAction();
```

Comment:
To make both condition branches look similar you could pass the table to the action constructor:

```java
removeFromCurrSet = new JmixRemoveFromSetAction(table);
```

Reply:
Done. Both actions accept their own ListComponent. Class cast was moved up a level.